zerver/migrations/0431_alter_and_migrate_realm_name_in_notifications.py

```diff
@@ -0,0 +1,93 @@
+from django.db.migrations.state import StateApps
+
+# We include a copy of this structure as it was at the time this
+# migration was merged, since future should not impact the migration.
+REALM_NAME_IN_NOTIFICATIONS_POLICY_AUTOMATIC = 1
+REALM_NAME_IN_NOTIFICATIONS_POLICY_ALWAYS = 2
+REALM_NAME_IN_NOTIFICATIONS_POLICY_NEVER = 3
+
+
+# The value of 'realm_name_in_notifications_policy' for those users
+# who have manually changed the value of 'realm_name_in_notifications' as 'true'
+# should be updated as 'Always', not 'Automatic'
+def update_realm_name_in_notifications_policy_values(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    UserProfile = apps.get_model("zerver", "UserProfile")
+    userprofiles = UserProfile.objects.filter(realm_name_in_notifications=True)
+    userprofiles.update(
+        realm_name_in_notifications_policy=REALM_NAME_IN_NOTIFICATIONS_POLICY_ALWAYS
+    )
+
+
+def reverse_code(apps: StateApps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    UserProfile = apps.get_model("zerver", "UserProfile")
+    userprofiles = UserProfile.objects.filter(
+        realm_name_in_notifications_policy=REALM_NAME_IN_NOTIFICATIONS_POLICY_ALWAYS
+    )
+    userprofiles.update(realm_name_in_notifications=True)
+
+
+def update_realm_name_in_notifications_policy_values_for_realm_user_default(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    RealmUserDefault = apps.get_model("zerver", "RealmUserDefault")
+    realmuserdefaults = RealmUserDefault.objects.filter(realm_name_in_notifications=True)
+    realmuserdefaults.update(
+        realm_name_in_notifications_policy=REALM_NAME_IN_NOTIFICATIONS_POLICY_ALWAYS
+    )
+
+
+def reverse_code_for_realm_user_default(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    RealmUserDefault = apps.get_model("zerver", "RealmUserDefault")
+    realmuserdefaults = RealmUserDefault.objects.filter(
+        realm_name_in_notifications_policy=REALM_NAME_IN_NOTIFICATIONS_POLICY_ALWAYS
+    )
+    realmuserdefaults.update(realm_name_in_notifications=True)
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("zerver", "0430_fix_audit_log_objects_for_group_based_stream_settings"),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name="realmuserdefault",
+            name="realm_name_in_notifications_policy",
+            field=models.PositiveSmallIntegerField(
+                default=REALM_NAME_IN_NOTIFICATIONS_POLICY_AUTOMATIC
+            ),
+        ),
+        migrations.RunPython(
+            update_realm_name_in_notifications_policy_values_for_realm_user_default,
+            reverse_code=reverse_code_for_realm_user_default,
+            elidable=True,
+        ),
+        migrations.RemoveField(
+            model_name="realmuserdefault",
+            name="realm_name_in_notifications",
+        ),
```

Comment:
It doesn't matter when this is run in a single transaction, which is the default, but it might be better to do both the `RemoveField` operations at the end, after the other steps have completed.

Reply:
Done